test/cypress/integration/widgets/filters_spec.ts

```diff
@@ -243,4 +269,433 @@ describe('Filters widget', () => {
+                const addModalErrors = [...addModalOnlyErrors, ...commonModalErrors];
+                addModalErrors.forEach(verifyErrorHandling);
+            });
+        });
+
+        it('when editing existing filter', () => {
+            openEditFilterModal();
+            cy.get('.modal').within(() => {
+                const editModalErrors = commonModalErrors;
+                editModalErrors.forEach(verifyErrorHandling);
+            });
+        });
+
+        it('when cloning existing filter', () => {
+            const cloneModalOnlyErrors: FilterModalError[] = [
+                {
+                    name: 'reserved filter ID provided',
+                    prepare: () => {
+                        getFilterIdInput().clear().type('csys-invalid');
+                    },
+                    verify: () => cy.contains('All filters with a `csys-` prefix are reserved for internal use'),
+                    clean: () => {}
+                }
+            ];
+
+            openCloneFilterModal();
+            cy.get('.modal').within(() => {
+                const cloneModalErrors = [...cloneModalOnlyErrors, ...commonModalErrors];
+                cloneModalErrors.forEach(verifyErrorHandling);
+            });
+        });
+    });
+
+    describe('should allow to define all kinds of filter rules', () => {
+        const testPrefix = 'filters_test_form';
+        const blueprintId = `${testPrefix}_blueprint`;
+        const deploymentId = `${testPrefix}_deployment`;
+
+        before(() => {
+            cy.deleteDeployments(testPrefix)
+                .deleteBlueprints(testPrefix)
+                .uploadBlueprint('blueprints/empty.zip', blueprintId)
+                .deployBlueprint(blueprintId, deploymentId)
+                .setLabels(deploymentId, [
+                    { os: 'linux' },
+                    { os: 'windows' },
+                    { infra: 'aws' },
+                    { infra: 'gcp' },
+                    { infra: 'azure' }
+                ])
+                .deleteSites(testPrefix)
+                .createSites([
+                    { name: `${testPrefix}_Cracow` },
+                    { name: `${testPrefix}_Warsaw` },
+                    { name: `${testPrefix}_TelAviv` }
+                ]);
+        });
+
+        function selectRuleRowType(ruleRowType: FilterRuleRowType) {
+            withinLastRuleRow(() => {
+                cy.get('div[name="ruleRowType"]').click();
+                cy.get(`div[option-value="${ruleRowType}"]`).click();
+            });
+        }
+
+        function selectRuleOperator(operator: FilterRuleOperator) {
+            withinLastRuleRow(() => {
+                cy.get('div[name="ruleOperator"]').click();
+                cy.get(`div[option-value="${operator}"]`).click();
+            });
+        }
+
+        function selectRuleAttributeValues(
+            ruleRowType: FilterRuleRowType,
+            values: string[],
+            newValues = [] as string[],
```

Comment:
I didn't understand what's the use of `values` vs `newValues` :worried: Could you describe it, perhaps right in the code?

Alternatively, we can make the code/variable names clearer :smile: 

I see we're typing `values` in the search field, and then picking `newValues` from it. Maybe we should rename the parameters to

```suggestion
            valuesToType: string[],
            newValuesToAdd = [] as string[],
```

Reply:
Another option would be to avoid accepting 2 separate arrays, and accept an array of objects of the following shape instead:

```ts
{
  value: string;
  isNew?: boolean;
}
```

Then, we co-locate whether some value should be new or it's an existing value :smile:

Reply:
Good idea with passing array of objects, will implement.